Add --preview option

Jack12816 opened the pull request:
This feature adds a preview rectangle which shows the current hovered
color at the left bottom corner in a 50x50 pixel area.

Demo video: https://s3-eu-west-1.amazonaws.com/jack12816/public/videos/vid-2017-08-31-131752.webm

Jack12816 committed:
Add --preview option

This feature adds a preview rectangle which shows the current hovered
color at the left bottom corner in a 50x50 pixel area.

Ancurio commented:
Thanks, but I'm a bit confused by why you changed the formatting from tabs to spaces in a lot of lines, and made other changes that have nothing to do with the feature? It makes it a bit hard to compare what actually changed.

Also you added dependencies on Xcomposite and Xfixes, that should be added to the README.

Jack12816 commented:
Eh yeah, I run uncrustify over the `main.c` file. I refactored the `main()` function and added the preview feature. As requested I reworked the readme and improved the `Makefile`. 

It's kind of "bad practice" in a PR, but the boy scout rule (Always leave the campground cleaner than you found it.) was more important to me here. I recommend to review it not side-by-side and read it completely again. 

Sorry for the trouble. :)

Ancurio commented:
Feel free to repoen when you have addressed the stated issues. I saw that you also cleaned up the Readme; I'm not opposed to that, but please open a separate PR for it. Thanks.
